src/passkey_child/passkey_child_assert.c

```diff
@@ -34,6 +34,67 @@
 
 #include "passkey_child.h"
 
+errno_t
+get_assert_user_id(TALLOC_CTX *mem_ctx, fido_assert_t *assert,
+                   unsigned char **_user_id)
+{
+    TALLOC_CTX *tmp_ctx = NULL;
+    size_t assert_number;
+    size_t user_id_len;
+    const unsigned char *user_id;
+    char *b64_user_id;
+    errno_t ret;
+
+    tmp_ctx = talloc_new(NULL);
+    if (tmp_ctx == NULL) {
+        ERROR("talloc_new() failed\n");
+        return ENOMEM;
+    }
+
+    assert_number = fido_assert_count(assert);
```

Comment:
@LDVG can `fido_assert_count()` return a value higher than 1?

Reply:
If your authenticatorGetAssertion request has an empty allow list, yes. See [CTAP 2.1 § 6.2.2  authenticatorGetAssertion Algorithm, step 12](https://fidoalliance.org/specs/fido-v2.1-ps-20210615/fido-client-to-authenticator-protocol-v2.1-ps-20210615.html#sctn-getAssert-authnr-alg).

As far as I can tell, [you're planning on always providing a Credential ID](https://github.com/SSSD/sssd/blob/dad2f17a70d6dcf797620dfd89f5effe0fffd5af/src/passkey_child/passkey_child_assert.c#L179)? As such, you'll always receive at most 1 credential back.

Reply:
> As far as I can tell, [you're planning on always providing at least one Credential ID](https://github.com/SSSD/sssd/blob/dad2f17a70d6dcf797620dfd89f5effe0fffd5af/src/passkey_child/passkey_child_assert.c#L179)? As such, you'll always receive at most 1 assertion back (per CTAP 2.1 § 6.2.2, step 11).

Yes, and that's what I wanted to hear back.

Feel free to review this PR as it is ready.